### PR TITLE
[backend/frontend] Fix removing a team from the context doesn't work

### LINKED ISSUE #1521: Removing a team from the context doesn't work
## Description

When I have a team assigned to some injects and I "remove it from the context" the team is still assigned to the injects 

## Environment

Testing 

## Reproducible Steps

- Go to https://testing.obas.staging.filigran.io/admin/exercises/85eb1560-3289-437e-835f-bd4cf6183ecd/definition
- Remove one of the team from the context 
image

- Look at the injects
- The team is still assigned
image

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
Closes #1521 

### Proposed changes

*
*

### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/1521
*

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/service/ScenarioService.java [hunk 1]
```diff
@@ -440,13 +441,14 @@ public Iterable<Team> addTeams(@NotBlank final String scenarioId, @NotNull final
...
+  @Transactional(rollbackFor = Exception.class)
   public Iterable<Team> removeTeams(@NotBlank final String scenarioId, @NotNull final List<String> teamIds) {
-    Scenario scenario = this.scenario(scenarioId);
-    List<Team> teams = scenario.getTeams().stream().filter(team -> !teamIds.contains(team.getId())).toList();
-    scenario.setTeams(new ArrayList<>(teams));
-    this.updateScenario(scenario);
+    // Remove teams from exercise
+    this.scenarioRepository.removeTeams(scenarioId, teamIds);
     // Remove all association between users / exercises / teams
     teamIds.forEach(this.scenarioTeamUserRepository::deleteTeamFromAllReferences);
```
Comment (MarineLeM) [inline openbas-api/src/main/java/io/openbas/service/ScenarioService.java:449]: Same question as above

File: openbas-api/src/test/java/io/openbas/service/ExerciseServiceTest.java [hunk 2]
```diff
@@ -1,81 +1,138 @@
...
+  }
+
+  @DisplayName("Should remove team from exercise")
+  @Test
+  void testRemoveTeams() {
+    // -- PREPARE --
+    User user = getUser();
+    User userSaved = this.userRepository.saveAndFlush(user);
+    USER_ID = userSaved.getId();
+    Team team = getTeam(user);
```
Comment (MarineLeM) [inline openbas-api/src/test/java/io/openbas/service/ExerciseServiceTest.java:114]: should be `Team team = getTeam(userSaved);`  ?

File: openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java [hunk 3]
```diff
@@ -110,4 +127,36 @@ void createNewContextualTeamsDuringScenarioDuplication(){
...
     }
+
+    @DisplayName("Should remove team from scenario")
+    @Test
+    void testRemoveTeams() {
+        // -- PREPARE --
+        User user = getUser();
+        User userSaved = this.userRepository.saveAndFlush(user);
+        USER_ID = userSaved.getId();
+        Team team = getTeam(user);
```
Comment (MarineLeM) [inline openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java:138]: same question as above ?

File: openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseService.java [hunk 4]
```diff
@@ -4,10 +4,8 @@
 import com.fasterxml.jackson.databind.node.ArrayNode;
```
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseService.java:4]: openbas-api/src/main/java/io/openbas/rest/exercise/service ?
Regarding the best practices on notion

File: openbas-api/src/test/java/io/openbas/service/ExerciseServiceTest.java (hunk 2 above)
Comment (RomuDeuxfois) [reply openbas-api/src/test/java/io/openbas/service/ExerciseServiceTest.java:118]: Good catch

File: openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseService.java [hunk 5]
```diff
@@ -407,4 +407,17 @@ private void getObjectives(Exercise duplicatedExercise, Exercise originalExercis
...
     }
 
+    // -- TEAMS --
+
+    @Transactional(rollbackFor = Exception.class)
+    public Iterable<Team> removeTeams(@NotBlank final String exerciseId, @NotNull final List<String> teamIds) {
+        // Remove teams from exercise
+        this.exerciseRepository.removeTeams(exerciseId, teamIds);
+        // Remove all association between users / exercises / teams
+        teamIds.forEach(exerciseTeamUserRepository::deleteTeamFromAllReferences);
```
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseService.java]: Well seen

**Comment by RomuDeuxfois** [comment]
Need to be retest

**Comment by RomuDeuxfois** [comment]
All good :)
Ready for a second round

**Review by MarineLeM — APPROVED** [review]

**Pull request merged**